fasm/model.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2017-2020  The SymbiFlow Authors.
+#
+# Use of this source code is governed by a ISC-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/ISC
+#
+# SPDX-License-Identifier: ISC
+
+from collections import namedtuple
+import enum
+
+
+# Number format used for a FASM value.
+class ValueFormat(enum.Enum):
+    PLAIN = 0  # 42
+    VERILOG_DECIMAL = 1  # 8'd42
+    VERILOG_HEX = 2  # 8'h2a
+    VERILOG_BINARY = 3  # 8'b00101010
+    VERILOG_OCTAL = 4  # 8'o52
+
+
+# Python version of a SetFasmFeature line.
```

Comment:
This should all be docstrings. https://stackoverflow.com/questions/1606436/adding-docstrings-to-namedtuples

Reply:
Wrapping a tuple in a class to add a docstring seems tedious, and probably adds runtime overhead.
@litghost What do you think about this?

Reply:
That page lists multiple ways to add a docstring to namedtuples.

Reply:
Done.